internal/resources/gsi.go

```diff
@@ -234,6 +234,20 @@ This will automatically be retried in the background.  Please run "terraform app
 			),
 		)
 
+		return
+	case internalerrors.ErrGatewayTimeoutForIndexDDL:
+		resp.Diagnostics.AddWarning(
+			"Long index build time",
+			fmt.Sprintf(
+				`Build of index %s in %s.%s.%s did not complete in time.  This maybe due to non-deferred builds of a large index,
+in which case it is recommended to use deferred builds.  Please monitor the index and once its ready, then refresh state.`,
```

Comment:
how can they monitor the index?

Reply:
updated the warning to inform user to monitor with query_index datasource.